.github/workflows/test.yml

```diff
@@ -53,9 +54,29 @@ jobs:
         uses: actions/cache@v2
         with:
           path: .tox
-          key: tox-cache-${{ matrix.python-version }}-${{ matrix.package }}-${{ matrix.os }}-${{ hashFiles('tox.ini', 'dev-requirements.txt') }}
+          key: tox-cache-${{ env.RUN_MATRIX_COMBINATION }}-${{ hashFiles('tox.ini', 'dev-requirements.txt') }}
       - name: run tox
-        run: tox -f ${{ matrix.python-version }}-${{ matrix.package }}
+        run: tox -f ${{ matrix.python-version }}-${{ matrix.package }} -- --benchmark-json=${{ env.RUN_MATRIX_COMBINATION }}-benchmark.json
+      - name: Find and merge benchmarks
+        # TODO: Add at least one benchmark to every package type to remove this
+        if: matrix.package == 'sdkextension'
+        run: jq -s '.[0].benchmarks = ([.[].benchmarks] | add) | if .[0].benchmarks == null then null else .[0] end' **/**/tests/*${{ matrix.package }}*.json > output.json
+      - name: Report on benchmark results
+        # TODO: Add at least one benchmark to every package type to remove this
+        if: matrix.package == 'sdkextension'
+        uses: rhysd/github-action-benchmark@v1
+        with:
+          name: OpenTelemetry Python Benchmarks - Python ${{ env[matrix.python-version ]}} - ${{ matrix.package-group }}
+          tool: pytest
+          output-file-path: output.json
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          # Alert with a commit comment on possible performance regression
+          alert-threshold: 200%
+          comment-always: true
+          fail-on-alert: true
```

Comment:
We could also keep [the conversation](https://github.com/open-telemetry/opentelemetry-python/pull/1443/files#r540463014) on https://github.com/open-telemetry/opentelemetry-python/pull/1443

> @aabmass @toumorokoshi  If we fail the test, and merge it into master, it will **not** publish the commit to `gh-pages` because the test failed.

> So the question is, is there a case where we would merge a PR which fails performance tests? If yes, then the maintainers who can merge would just have to flip to flag to `false` for the push to `gh-pages` to succeed and flip the flag to back to `true` later to fail subsequent PRs which hurt performance.

> The other option is to not fail. In that case the push to `gh-pages` will always succeed, but the CI will pass even when performance gets bad. (But we will still have a comment that says performance got worse!)

> cc @lzchen @codeboten

Reply:
Resolved that performance regressions should be merge blocking so we won't have a problem here!